test/support/module_based_registry.ex

```diff
@@ -13,3 +13,45 @@ defmodule TestRegistry2 do
     {:ok, Keyword.put(options, :members, [:init_test_1, :init_test_2])}
   end
 end
+
+defmodule TestRegistry3 do
+  use Horde.Registry
+  @base_opts [keys: :unique, name: __MODULE__]
+
+  def base_opts, do: @base_opts
+
+  def init(options) do
+    {:ok, Keyword.put(options, :members, [:init_test_1, :init_test_2])}
+  end
+
+  def child_spec(args) do
+    %{
+      id: args[:custom_id],
+      start: {__MODULE__, :start_link, [Keyword.merge(@base_opts, args)]},
+      restart: :transient,
+      type: :supervisor
+    }
+  end
+end
+
+defmodule TestRegistry4 do
+  use Horde.Registry,
+    restart: :transient,
+    # value provided to differ from default :supervisor option
+    type: :worker
```

Comment:
Demonstrate configurability of `child_spec` through use of options like [GenServer](https://github.com/elixir-lang/elixir/blob/c76c5d6d188bbb787d6afefab74a31080695d1f2/lib/elixir/lib/gen_server.ex#L128-L130)